### DISCUSSION TITLE
Slack or Discord?

### DISCUSSION THREAD

**Discussion by bjchambers** [kaskada-ai/kaskada#180] [General]
Thoughts on whether it would be helpful to have a Slack or Discord group for realtime discussions? Any preferences on which one?

**Comment by bjchambers** [comment]
Looking around, it seems mixed.

https://thenewstack.io/discord-vs-slack-which-is-right-for-your-team/

* Slack tends to have better support for text (likely to be the most commonly used)
* Discord provides unlimited history, which has benefits.

Looking at some other projects
* Apache Software Foundation (ASF) runs a slack instance, so many Apache projects show up there
* arrow-rs has a discord channel (https://github.com/apache/arrow-rs) (in addition to a slack channel on the ASF instance)

**Reply by kerinin** [reply]
From the first article: "In Slack’s philosophy, a person who joins the workspace is a member invited by a trusted person."

Some more community examples:
* Slack: Flink, Pulsar, Feast, RAPIDS, Kudu
* Discord: DuckDB, Polars, PQRL
* Discourse: Dask
* None: Spark, Beam, Calcite

It looks like smaller, newer projects are more likely to use Discord. Larger, older, projects backed by bigger organizations seem more likely to use Slack.

**Reply by jbellis** [reply]
ASF provides full enterprise Slack to its projects which is a nice value-add, but in general newer projects started preferring discord as more people realized how crippling Slack's free message retention limitations are.

**Reply by bjchambers** [reply]
Adding:

* arrow-rs: ASF Slack *and* Discord
* pandas: Has a slack, but only for contributors (not user questions). https://pandas.pydata.org/docs/dev/development/community.html?highlight=slack#community-slack
* asciidoctor: https://zulip.com/case-studies/asciidoctor/
* rust: https://zulip.com/case-studies/rust/

**Comment by CharnaParkey** [comment]
Yes! I'd love a place for conversation with this community in real time that isn't public and associated with my github account for more casual conversations. 

Here's my pro con for a few:

Slack - I already use it for work so I would see messages without needing to have another app open. Slack is a name people know and trust. Chances are, a good proportion of community members will be familiar with it. This may help initial adoption. But it will need to be paid eventually as we grow, and there is no really effective way to moderate.

Discord - unlimited message history, built in moderation features. In Discord, there’s no easy way to create a new profile each time you join a new community. Instead, you use your one-and-only profile across all groups. This might be problematic for those who want a degree of separation between the different parts of their lives. 

Discourse - 100% Open source and customizable, more forum based conversations. But our community isn't likely big enough to need this level of organization yet.

**Reply by bjchambers** [reply]
That makes sense! It feels like Discourse is somewhat duplicative with GitHub discussions. It seems like it could confuse the question of where to go to have a forum-based conversation.

So it seems like Slack vs. Discord is the decision to make here.

**Reply by bjchambers** [reply]
Looking into this more, I also saw people suggesting `zulip`. It is free for open source projects, and is in use by both Rust (the compiler and language teams) and AsciiDoctor (which we use for rendering the site), both of which are pretty sizeable.

They had used both Slack (moved off it due to the message limit and cost) and Discord (moved off it because it wasn't accessible in some regions, and didn't make it easy to embed code and such).

See for instance https://github.com/engula/engula/issues/190.

**Reply by CharnaParkey** [reply]
I personally hesitate to have yet another messaging app over slack and discord since it's another source of notifications but I'm up for trying it out and evaluating if it's working for the community again soon

**Reply by jbellis** [reply]
Discord vs Zulip

![image](https://user-images.githubusercontent.com/42158/228961309-828cecf8-d0ae-47b3-9fd9-463e8002ed75.png)

**Reply by jordanrfrazier** [reply]
I could definitely see benefits to having both discord and slack. Slack feels like the default for professionals interested in asking questions during work hours. Discord I'll have open in off-hours, where I'm randomly a part of discords for hobbies/random interests. I see that being attractive to people that work on side-projects, the tech tinkerers. There's certainly ways to avoid spreading out information across multiple platforms too - we can write a discord bot that relays messages to the matching slack channels, and likely vice versa.

**Reply by jbellis** [reply]
I would strongly prefer not to go with an "all of the above" approach, and I have never seen a bridge or relay approach work well.

**Comment by pmcfadin** [comment]
Thanks for the invite Jonathan :p 

For DataStax DevRel we use Discord for our community work. https://discord.gg/9js9vjmsK8 

Here are a few reasons we've decided to go this route.
 - Easy for users to access. Slack requires a lot of hoops like invites.
 - Great community features like onboarding new users. Auto-moderation. An XP system to level up to more privileges
 - Tons of useful bots like MEE6, Community verification and Orbit.
 - The user stats on Discord are way better than Slack could ever hope for. 

Things to consider.
 - Discord requires care and feeding. It's an active platform, and you need to spend the time to make it awesome. Great example of an awesome discord: https://discord.gg/Rqqngehh
 - The interface is different than Slack. That can cause a reaction in the most conservative Slack users. 
 - Since discords are searchable, you will get spam. Get the anti-spam bots figured out.
 - Sometimes the basic-ness of Slack is good enough.

**Comment by FelixGV** [comment]
Regarding unlimited history, there is a way to skirt around the limitations of a free Slack instance, somewhat. By integrating the Slack with Linen, there is a public/searchable archive of the Slack content available on the web. Not as nice as having it right there in the chat app, but at least the conversations aren't totally lost... here's an example of a Linen instance for the Venice project's Slack: http://linen.venicedb.org
